src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComInterfaceMarshaller.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+
+namespace System.Runtime.InteropServices.Marshalling
+{
+    /// <summary>
+    /// COM interface marshaller using a StrategyBasedComWrappers instance
+    /// </summary>
+    /// <remarks>
+    /// This marshaller will always pass the <see cref="CreateObjectFlags.Unwrap"/> flag
+    /// to <see cref="ComWrappers.GetOrCreateObjectForComInstance(IntPtr, CreateObjectFlags)"/>.
+    /// </remarks>
+    [UnsupportedOSPlatform("android")]
+    [UnsupportedOSPlatform("browser")]
+    [UnsupportedOSPlatform("ios")]
+    [UnsupportedOSPlatform("tvos")]
+    [CLSCompliant(false)]
+    [CustomMarshaller(typeof(CustomMarshallerAttribute.GenericPlaceholder), MarshalMode.Default, typeof(ComInterfaceMarshaller<>))]
+    public static unsafe class ComInterfaceMarshaller<T>
+    {
+        public static void* ConvertToUnmanaged(T? managed)
+        {
+            if (managed == null)
+            {
+                return null;
+            }
+            if (!ComWrappers.TryGetComInstance(managed, out nint unknown))
+            {
+                unknown = StrategyBasedComWrappers.DefaultMarshallingInstance.GetOrCreateComInterfaceForObject(managed, CreateComInterfaceFlags.None);
+            }
+            return CastIUnknownToInterfaceType(unknown);
+        }
+
+        public static T? ConvertToManaged(void* unmanaged)
+        {
+            if (unmanaged == null)
+            {
+                return default;
+            }
+            return (T)StrategyBasedComWrappers.DefaultMarshallingInstance.GetOrCreateObjectForComInstance((nint)unmanaged, CreateObjectFlags.Unwrap);
+        }
+
+        private static readonly Guid? TargetInterfaceIID;
+
+        static ComInterfaceMarshaller()
+        {
+            if (StrategyBasedComWrappers.DefaultIUnknownInterfaceDetailsStrategy.GetIUnknownDerivedDetails(typeof(T).TypeHandle) is { } interfaceDetails)
```

Comment:
> `is { } interfaceDetails`

What is this?

Reply:
This is the suggested way to check "is not null and capture an object into a variable" from Roslyn. It's basically a typed "is this value a non-null instance with the listed (no) properties pattern matched". I'll change this to be an explicit typed pattern match instead of using this pattern.